### PR TITLE
NF-403 and NF-404 - file upload improvements

### PR DESCRIPTION
**Pull request by SteveSugden**
403 - start spinner animation and disable continue button immediately on file selection
404 - align spinner and "READY TO SUBMIT" tag further to the left to create more space for file name

### CODE REVIEW

**Review by rorticus0502 — APPROVED** [review]

File: app/assets/stylesheets/upload-loading.scss [hunk 1]
```diff
@@ -34,3 +34,7 @@
 .nidac-display-only-non-javascript {
   display: none;
 }
+
+.nidac-upload-filename {
+  width: 80%;
```
Comment (rorticus0502) [inline app/assets/stylesheets/upload-loading.scss:39]: what was the width for this element defaulting to?

Comment (crosslandwa) [reply app/assets/stylesheets/upload-loading.scss:39]: Can we achieve the same layout using https://design-system.service.gov.uk/styles/spacing/#width-override-classes??

Comment (SteveSugden) [reply app/assets/stylesheets/upload-loading.scss:39]: Yes!  `govuk-!-width-three-quarters` achieves the same thing.  Thanks

File: app/assets/javascripts/nidacfrontend.js [hunk 2]
```diff
@@ -10,4 +10,15 @@ if (window.history && window.history.replaceState && typeof window.history.repla
   window.history.replaceState(null, null, window.location.href);
 }
 
-
+// automatically submit document upload on file select
+function onFileSelect() {
+  try{
+    // NF-403 - hide the form controls, display the spinner and disable the 'Continue' button before submitting
+    document.getElementById("upload-form").className = "govuk-!-display-none";
+    document.getElementsByClassName("hidden-spinner-row").item(0).className = "govuk-summary-list__row";
```
Comment (crosslandwa) [inline app/assets/javascripts/nidacfrontend.js:18]: I think the following should work, and de-couples this functionality from summary list class names
```suggestion
    document.querySelector(".hidden-spinner-row").classList.remove("govuk-!-display-none");
```

File: app/assets/javascripts/nidacfrontend.js [hunk 3]
```diff
@@ -10,4 +10,15 @@ if (window.history && window.history.replaceState && typeof window.history.repla
   window.history.replaceState(null, null, window.location.href);
 }
 
-
+// automatically submit document upload on file select
+function onFileSelect() {
+  try{
+    // NF-403 - hide the form controls, display the spinner and disable the 'Continue' button before submitting
+    document.getElementById("upload-form").className = "govuk-!-display-none";
```
Comment (crosslandwa) [inline app/assets/javascripts/nidacfrontend.js:17]: For 
- consistency with the below suggestion
- avoiding issues with removing "other" classes on the form that might be required elsewhere...

```suggestion
    document.getElementById("upload-form").classList.add("govuk-!-display-none");
```

Comment (SteveSugden) [reply app/assets/javascripts/nidacfrontend.js:17]: Thanks for both suggestions.  Should work with IE10 according to https://www.w3schools.com/jsref/prop_element_classlist.asp 

I'll give it a go

File: app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/views/components/uploadSpinner.scala.html [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *@
+
+@()
+
+<span id="nidac-spinner" class="govuk-body ccms-loader"></span>
```
Comment (crosslandwa) [inline app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/views/components/uploadSpinner.scala.html:19]: The `id="nidac-spinner"` is only used by the acceptance tests which leaves them vulnerable to breaking if someone "cleaned up an unused ID"

Can the acceptance tests change from
```
val loadingAnimation = driver.findElement(By.id("nidac-spinner"))
```

to something like
```
val loadingAnimation = driver.findElement(By.class("ccms-loader"))
```

i.e. using the same selector that is used to apply the styles

Comment (SteveSugden) [reply app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/views/components/uploadSpinner.scala.html:19]: You are correct - I added the id specifically for the ATs.    But the ATs can always be broken by changes to the FE code.  Someone might decide that the class name `ccms-loader` doesn't make sense and change it.

Comment (crosslandwa) [reply app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/views/components/uploadSpinner.scala.html:19]: Agreed, its brittle either way

File: app/assets/stylesheets/upload-loading.scss (hunk 1 above)
Comment (SteveSugden) [reply app/assets/stylesheets/upload-loading.scss]: Changed

File: app/assets/javascripts/nidacfrontend.js (hunk 2 above)
Comment (SteveSugden) [reply app/assets/javascripts/nidacfrontend.js]: Done

File: app/assets/javascripts/nidacfrontend.js (hunk 3 above)
Comment (SteveSugden) [reply app/assets/javascripts/nidacfrontend.js]: Done and tested on IE11 using BrowserStack

**Review by crosslandwa — APPROVED** [review]

**Pull request merged**